pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -0,0 +1,427 @@
+package esgateway
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/dns"
+	"github.com/tigera/operator/pkg/render"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+)
+
+const (
+	EsGatewayName                    = "tigera-es-gateway"
+	EsGatewayServiceAccountName      = "tigera-es-gateway"
+	EsGatewayRole                    = "tigera-es-gateway"
+	EsGatewaySecretKeyName           = "key"
+	EsGatewaySecretCertName          = "tls.crt"
+	EsGatewayVolumeName              = "tigera-es-gateway-certs"
+	EsGatewayElasticServiceName      = "tigera-secure-es-http"
```

Comment:
Moving these constants here make it so that every component now needs to know of the existence of the gateway, when they really don't. Logically the gateway is an invisible component, and the gateway url and certs should really be exposed to them as Elasticsearch items. 

The names are the same, the _only_ item that should know that there is a gateway is the logstorage controller, and the Gateway and Elasticsearch render items should be configurable enough the the logstorage controller can configure them accordingly.

I would keep these parameters in the package that they were in, and pass the values into the gateway and elasticsearch components. That way we can still keep all the other components blind to the fact that there is a gateway, to them there's only Elasticsearch.